### DISCUSSION TITLE
Skip if exists

### DISCUSSION THREAD

**Discussion by jidanni** [curl/curl#11012] [Ideas]
Curl has no exact equivalent for the shell's `test -e`.
No way to say 
> if the destination file already exists locally,
> then don't even probe the network about it.
> Simply skip to the next file.

Sure, you might say "Well, then just do
> test -e file || curl ... -o file"

Yes, but let's say we want to get 1000 files. We could write a config file to do that.
But we need to now create the config file dynamically, as there is no test that we can insert.

Sure you might say
> Well, why would you not want to take advantage of -C and curl will as a bonus check that your file is in good shape.

Well, we might know they are all in good shape.

**Comment by tobwen** [comment]
Would be nice to have a switch like `--skip-existing`, which also works with globbing.

**Comment by Manwithpants812** [comment]
Please add this. It's a very basic feature.

**Comment by danoli3** [comment]
Please cURL get with the times we need this command optimisation.

Here is an idea for implementation: 
```
Local=$(wc -c < $OUT_FILE.zip)
Remote=$(curl -sI $URL | awk '/Content-Length/ {sub("\r",""); print $2}')
if [ $Local != $Remote ]; then
echo "Run the cURL"
else
echo "Same size. Do not run the cURL"
fi
```

Can also use server time but I think content length is a pretty good indicator of change
```
REMOTE="$URL"
LOCAL="$OUT_FILE"

modified=$(curl --silent --head $REMOTE| \
             awk '/^Last-Modified/{print $0}' | \
             sed 's/^Last-Modified: //')
remote_ctime=$(date --date="$modified" +%s)
local_ctime=$(stat -c %z "$LOCAL")
local_ctime=$(date --date="$local_ctime" +%s)

if [ $local_ctime -lt $remote_ctime ]; then
echo "Run the cURL"
else 
echo "Time modified has not changed on server . Do not run the cURL"
fi
```

**Reply by bagder** [reply]
@danoli3 you seem to be asking for `-z` which has been around for a while. Usually `--etag-compare` is slightly better.

**Reply by danoli3** [reply]
What cURL version supports ?!! w000

**Reply by bagder** [reply]
- `-z` (`--time-cond`) was added in curl 5.8 (May 5 1999)
- etags support was added in curl 7.68.0 (Jan 8 2020)

**Reply by vszakats** [reply]
I think there is point that neither `-z` nor `--etag-compare` play well with globbing.

Would a parameterless `-z` flavour help, which automatically uses the current output filename as the reference?

For ETags, perhaps storing them with `--xattr` and retrieving them from there could help to make it globbing-friendly? (Though extended attribues are not available on all platforms.)

**Comment by bagder** [comment]
See  #13993
